frontend/src/Utils/Utils.tsx

```diff
@@ -167,7 +167,22 @@ export const mapSuccess = <T, U>(
 export const orElse = <T, U>(v: Validation<T>, e: () => U) =>
   v.tag === VTag.Success ? v.data : e();
 
+export const validateNullable = <T,>(x: T | null | undefined): Validated<T> =>
+  x === null || x === undefined ? failure(Error("Was null!")) : success(x);
+
+export const keepIf = <T,>(b: boolean, x: T) => (b ? x : null);
+
+export const transpose = <T,>(m: T[][]): T[][] =>
+  m.length === 0 ? m : m[0].map((_, i) => m.map((row) => row[i]));
+
+export const all = <T,>(x: (T | null)[]): T[] | null =>
+  x.reduce<T[] | null>(
+    (p, c) => (c === null ? null : p?.concat([c]) ?? null),
+    [],
+  );
+
 // A mechine is busy if all gpus are in use
+// This utility really does not fit in this file...
```

Comment:
move it then

Reply:
https://github.com/gpuctl/gpuctl/pull/221/commits/d6b6eda6d34337f03cc9a13196529f934ebfddbd